pydoctor/epydoc2stan.py

```diff
@@ -305,8 +312,35 @@ def handle_param(self, field: Field) -> None:
         name = self._handle_param_name(field)
         if name is not None:
             self.add_info(self.parameter_descs, name, field)
+            if name in self.types:
+                return
+
+            # Figure out if the parameter might exist despite not being found
+            # in this documentable's code, warn if not.
+            source = field.source
+            if source is not self.obj:
+                # Docstring is inherited, so it may not represent this exact method.
+                return
+            if isinstance(source, model.Class):
+                if None in source.baseobjects:
+                    # Class has a computed base class, which could define parameters
+                    # we can't discover.
+                    # For example, this class might use
+                    # L{twisted.python.components.proxyForInterface()}.
+                    return
+                if name in source.constructor_params:
+                    # Constructor parameters can be documented on the class.
+                    return
+            field.report('Documented parameter "%s" does not exist' % (name,))
     handle_arg = handle_param
-    handle_keyword = handle_param
+
+    def handle_keyword(self, field: Field) -> None:
+        name = self._handle_param_name(field)
+        if name is not None:
+            # TODO: How should this be matched to the type annotation?
```

Comment:
There are two functions in Twisted that use `@keyword`, but neither of them have type annotations at the moment, so I think it's fine to resolve this TODO later.

Reply:
Do we want to support `@keyword` ... can't we just use `@param` ? :)

I see that in the previous code keyword was the same as param... so maybe deprecated `keyword` and just use `@param` everywhere.

Reply:
The annotation would be useful only if all keywords have the same type: if for example one keyword is a `str` and another keyword is an `int`, the annotation will be `object` or `Any` or `Union[int, str]`, which isn't very helpful to the end user. So the author will have to use `type` fields to document the types of the individual keywords.

Reply:
> Do we want to support `@keyword` ... can't we just use `@param` ? :)
> 
> I see that in the previous code keyword was the same as param... so maybe deprecated `keyword` and just use `@param` everywhere.

The purpose of `@keyword` is to document keywords that can be captured by the `**kwargs` parameter. I think it's useful to have a separate field for those, since otherwise we wouldn't be able to warn about `@param` being used with a non-existing parameter name. That warning led to several documentation fixes in Twisted.